src/main/java/seedu/address/logic/commands/EditEventCommand.java

```diff
@@ -0,0 +1,195 @@
+import seedu.address.model.event.StartDate;
+import seedu.address.model.event.StartTime;
+
+/**
+ * Edits the details of an existing event in the application.
+ */
+public class EditEventCommand extends Command {
+
+    public static final String COMMAND_WORD = "editEvent";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the event identified "
+            + "by the index number used in the displayed event list. "
+            + "Existing values will be overwritten by the input values. \n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_EVENT_TITLE + "EVENT_TITLE]"
+            + "[" + PREFIX_START_DATE + "DATE]"
+            + "[" + PREFIX_START_TIME + "TIME]"
+            + "[" + PREFIX_PURPOSE + "PURPOSE] \n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_EVENT_TITLE + "Banana Sale "
+            + PREFIX_PURPOSE + "20% discount on all bananas store-wide";
+
+    public static final String MESSAGE_EDIT_EVENT_SUCCESS = "Edited Event: %1$s";
+
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+
+    public static final String MESSAGE_DUPLICATE_EVENT = "This event already exists in the address book.";
+
+    private final Index index;
+
+    private final EditEventDescriptor editEventDescriptor;
+
+    /**
+     * Constructor of the EditEventCommand class
+     * @param index of the event to be edited in the filtered event list
+     * @param editEventDescriptor details to edit the event with
+     */
+    public EditEventCommand(Index index, EditEventDescriptor editEventDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editEventDescriptor);
+
+        this.index = index;
+        this.editEventDescriptor = new EditEventDescriptor(editEventDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Event> currentEventList = model.getFilteredEventList();
+
+        int zeroBased = index.getZeroBased();
+
+        if (zeroBased >= currentEventList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_EVENT_DISPLAYED_INDEX);
+        }
+
+        Event eventToEdit = currentEventList.get(zeroBased);
+        Event editedEvent = createEditedEvent(eventToEdit, editEventDescriptor);
+
+        if (!eventToEdit.isSameEvent(editedEvent) && model.hasEvent(editedEvent)) {
+            throw new CommandException(MESSAGE_DUPLICATE_EVENT);
+        }
+
+        model.setEvent(eventToEdit, editedEvent);
+        model.updateFilteredEventList(Model.PREDICATE_SHOW_ALL_EVENTS);
+        return new CommandResult(String.format(MESSAGE_EDIT_EVENT_SUCCESS, editedEvent));
+    }
+
+    /**
+     * Creates and returns a {@code Event} with the details of {@code eventToEdit} that is edited
+     * with details of {@code editEventDescriptor}
```

Comment:
May require full stop.
```suggestion
     * with details of {@code editEventDescriptor}.
```